### DISCUSSION TITLE
Edge config suitable for sensitive values?

### DISCUSSION THREAD

**Discussion by dmudro** [vercel/community#1131] [Help] [answered]
looking at the recently [privately released edge config](https://vercel.com/blog/edge-config-ultra-low-latency-data-at-the-edge)... would it be a suitable place to store sensitive values such as secrets?

the question stems from my ['runtime env vars' use case](https://github.com/orgs/vercel/discussions/1032).

**Comment by amyegan** [comment]
Good question @dmudro! The feature is still so new that I don't know for sure on this one, so I forwarded it to Edge Config team to get their input 🙏

**Reply by amyegan** [reply, marked as answer]
Sorry for the delay. Here's what the engineers had to say about the current state of the Edge Config beta for sensitive values:

> Env Vars are encrypted by us when we store them, but Edge Config contents are not. Env Vars are only transported in encrypted form unless you explicitly decrypt them, but Edge Config is transported in unencrypted form. However HTTPS encrypts the traffic anyhow. If an Edge Config token leaks, all content of that Edge Config can be read by whoever gains the token.
>
> To play it absolutely safe they could store a private key in their env vars and use that to encrypt the sensitive Edge Config contents on write and decrypt them when reading. That way all the points above are addressed.

**Reply by dmudro** [reply]
thanks! makes all sense - DYI edge config encryption (or parts of it).

**Comment by steamfeifei** [comment]
已收到   谢谢!!

**Comment by dmudro** [comment]
hey @amyegan, I wonder what's the situation around storing sensitive tokens (such as secrets) in edge config? would you be able to provide (updated) guidance or perhaps it's [the same answer](https://github.com/orgs/vercel/discussions/1131#discussioncomment-4392437)? thanks!

**Reply by amyegan** [reply]
Yes, the original answer is still the most secure option 🙂
